File: 0280-wiggle-sort/0280-wiggle-sort.cpp
class Solution {
public:
    void wiggleSort(vector<int>& nums) {
        for(int i = 0; i < nums.size()-1; i++){
            if((i % 2 == 0 && nums[i] > nums[i+1]) || (i % 2 == 1 && nums[i] < nums[i+1])){
                swap(nums[i], nums[i+1]); 
            }
        }

    }
};

File: 0280-wiggle-sort/0280-wiggle-sort.java
class Solution {
    public void wiggleSort(int[] nums) {
        //3 5 2 6 1 4 

        for(int i = 1; i < nums.length; i+=2){
            if(i + 1 < nums.length && nums[i] < nums[i+1]){
                int tmp = nums[i]; 
                nums[i] = nums[i+1]; 
                nums[i+1] = tmp; 
            }
        }

        for(int i = 0; i < nums.length; i+=2){
            if(i + 1 < nums.length && nums[i] > nums[i+1]){
                int tmp = nums[i]; 
                nums[i] = nums[i+1]; 
                nums[i+1] = tmp; 
            }
        }
    }


}